repository/src/main/java/org/apache/atlas/tasks/TaskRegistry.java

```diff
@@ -128,6 +127,38 @@ public List<AtlasTask> getInProgressTasks() {
         return ret;
     }
 
+    public List<AtlasTask> getInProgressTasksES() {
+        List<AtlasTask> ret = new ArrayList<>();
+        int size = 100;
+        int from = 0;
+            while(true) {
+                List statusClauseList = new ArrayList();
+                statusClauseList.add(mapOf("match", mapOf(TASK_STATUS, AtlasTask.Status.IN_PROGRESS.toString())));
+
+                Map<String, Object> dsl = mapOf("query", mapOf("bool", mapOf("must", statusClauseList)));
+                dsl.put("sort", Collections.singletonList(mapOf(Constants.TASK_CREATED_TIME, mapOf("order", "asc"))));
+                dsl.put("size", size);
+                dsl.put("from", from);
+                TaskSearchParams taskSearchParams = new TaskSearchParams();
+                taskSearchParams.setDsl(dsl);
```

Comment:
maintain static objects for the query & sort, for each request both should be constant
dsl can be initialised per request which will use static query & sort